src/components/date-picker/flatpickr-adapter/flatpickr-adapter.tsx

```diff
@@ -0,0 +1,180 @@
+     */
+    @Prop()
+    public language: Languages = 'en';
+
+    /**
+     * Emitted when the date picker value is changed.
+     */
+    @Event()
+    public change: EventEmitter<Date>;
+
+    private picker: Picker;
+    private flatPickrCreated: boolean = false;
+
+    private container: HTMLElement;
+
+    public componentWillLoad() {
+        switch (this.type) {
+            case 'date':
+                this.picker = new DateOnlyPicker(
+                    this.format,
+                    this.language,
+                    this.change
+                );
+                break;
+
+            case 'time':
+                this.picker = new TimePicker(
+                    this.format,
+                    this.language,
+                    this.change
+                );
+                break;
+
+            case 'week':
+                this.picker = new WeekPicker(
+                    this.format,
+                    this.language,
+                    this.change
+                );
+                break;
+
+            case 'month':
+                this.picker = new MonthPicker(
+                    this.format,
+                    this.language,
+                    this.change,
+                    translate
+                );
+                break;
+
+            case 'quarter':
+                this.picker = new QuarterPicker(
+                    this.format,
+                    this.language,
+                    this.change,
+                    translate
+                );
+                break;
+            case 'year':
+                this.picker = new YearPicker(
+                    this.format,
+                    this.language,
+                    this.change,
+                    translate
+                );
+                break;
+
+            case 'datetime':
+            default:
+                this.picker = new DatetimePicker(
+                    this.format,
+                    this.language,
+                    this.change
+                );
+                break;
+        }
+    }
+
+    public componentDidUpdate() {
+        if (this.flatPickrCreated) {
+            this.redrawFlatpickr();
+        } else {
+            this.createFlatpickr();
+        }
+        this.tryFixConfusingWidthBug();
+    }
+
+    private tryFixConfusingWidthBug() {
+        // Sometimes the datepickr renders with the width set to 1px.
+        // We've not been able to understand why but believe it has
+        // to do with the internal implementation of flatpickr.
+        // The fix below is an ugly fix that seems to solve this
+        // issue for us.
+        const ONE_SECOND = 1000;
+        const TEN_PIXELS = 10;
+        setTimeout(() => {
+            const { width } = this.container.getBoundingClientRect();
+            if (width < TEN_PIXELS) {
+                this.destroyFlatpickr();
+                this.createFlatpickr();
+            }
+        }, ONE_SECOND);
```

Comment:
Inside this timeout, we need to check if the portal is still open. Because if it's not, we're destroying a perfectly good flatpickr, and creating a new one in a circumstance where it's guaranteed to fail… 😆

Reply:
haha! That's true! Do you wanna push a fix for it? 🙏

Reply:
I pushed a fix for that. It turns out that the flatpickr element still has its width set when the portal is closed. So by looking at that element instead of the container we can decide if we should rerender.

Reply:
Ah, that sounds great! I'll test it again today 👍